libs/datastructures/include/hpx/datastructures/tuple.hpp

```diff
@@ -306,6 +331,18 @@ namespace hpx { namespace util {
                     typename detail::at_index<I, Ts...>::type> const&>(*this)
                     .value();
             }
+
+            HPX_HOST_DEVICE
+            operator std::tuple<Ts...>() const &
+            {
+                return std::make_tuple(get<Is>()...);
+            }
+
+            HPX_HOST_DEVICE
+            operator std::tuple<Ts...>() &&
+            {
+                return std::make_tuple(std::move(get<Is>())...);
+            }
```

Comment:
Didn't we at some point have also overloads for `&` and `const &&` ? Don't we need those?

Reply:
Good point... don't know why I lost them ... was probably due to resolving the merge conflicts :/